### ISSUE TITLE
The BaseClient _customHeaders assignment is not thread safe

### ISSUE THREAD

**Issue by mateuszligeza** [MiXTelematics/MiX.Integrate.Api.Client#28]
When multiple instances of api clients are created at the same time, the following error gets thrown
![image](https://user-images.githubusercontent.com/1456749/76262552-6236ae80-62a8-11ea-9615-0def654d2204.png)
Issue is caused by the bug in `MiX.Integrate.API.Client.Base.BaseClient`
```
//Build custom headers
_customHeaders = new Dictionary<string, string>();
_customHeaders.Add("X-MiX-LibraryInfo", $"Name: {assemblyName.Name}, Version: {assemblyName.Version.ToString()}, FileVersion: {assemblyFileVersion}, TargetFramework: {targetFramework}, Runtime: {imageRuntimeVersion}, EntryAssembly: {entryAssemblyNameHash}");
```
and can be fixed by inlining the dictionary creation
```
_customHeaders = new Dictionary<string, string>
{
    {
        "X-MiX-LibraryInfo",
        $"Name: {assemblyName.Name}, Version: {assemblyName.Version.ToString()}, FileVersion: {assemblyFileVersion}, TargetFramework: {targetFramework}, Runtime: {imageRuntimeVersion}, EntryAssembly: {entryAssemblyNameHash}"
    }
};
```
or changing the base constructor to be static
```
static BaseClient()
{
	//This assembly
	var assembly = typeof(BaseClient).GetTypeInfo().Assembly;
	var assemblyName = assembly.GetName();
	var version = assemblyName.Version;
	var imageRuntimeVersion = assembly.ImageRuntimeVersion;
	var assemblyFileVersionAttribute = (AssemblyFileVersionAttribute)assembly.GetCustomAttributes(typeof(AssemblyFileVersionAttribute)).FirstOrDefault();
	var targetFrameworkAttribute = (TargetFrameworkAttribute)assembly.GetCustomAttributes(typeof(TargetFrameworkAttribute)).FirstOrDefault();
	var assemblyFileVersion = (assemblyFileVersionAttribute != null) ? assemblyFileVersionAttribute.Version : "NULL";
	var targetFramework = (targetFrameworkAttribute != null) ? targetFrameworkAttribute.FrameworkName : "NULL";
	//Calling assembly 
	Assembly entryAssembly = Assembly.GetEntryAssembly();
	var entryAssemblyName = entryAssembly != null ? entryAssembly.GetName().Name : "NA";
	var algorithm = System.Security.Cryptography.SHA256.Create();
	var entryAssemblyNameHash = entryAssemblyName == "NA" ? "NA" : BitConverter.ToString(algorithm.ComputeHash(Encoding.UTF8.GetBytes(entryAssemblyName))).Replace("-", "");
	algorithm.Dispose();
	//Build custom headers
	_customHeaders = new Dictionary<string, string>();
	_customHeaders.Add("X-MiX-LibraryInfo", $"Name: {assemblyName.Name}, Version: {assemblyName.Version.ToString()}, FileVersion: {assemblyFileVersion}, TargetFramework: {targetFramework}, Runtime: {imageRuntimeVersion}, EntryAssembly: {entryAssemblyNameHash}");
}
```

**Comment by AlainH** [comment]
Resolved in #30

**Comment by mateuszligeza** [comment]
Thank you :)
